numba/datamodel/arginfo.py

```diff
@@ -0,0 +1,118 @@
+from __future__ import print_function, absolute_import
+
+from collections import deque
+
+
+class ArgInfo(object):
+    """
+    Compute the position for each high-level typed argument.
+    It flattens every composite argument into primitive types.
```

Comment:
Can you explain in which circumstances the flattening is useful?
(CUDA not supporting nested structs, perhaps? I don't remember)
Also, "ArgInfo" doesn't sound very descriptive :-)